database/mysql/mysql_user.py

```diff
@@ -307,19 +317,28 @@ def privileges_revoke(cursor, user,host,db_table,grant_option):
     query = ["REVOKE ALL PRIVILEGES ON %s" % mysql_quote_identifier(db_table, 'table')]
     query.append("FROM %s@%s")
     query = ' '.join(query)
-    cursor.execute(query, (user, host))
+    try:
+        cursor.execute(query, (user, host))
+    except Exception, e:
+        raise Exception("%s. Query=\"%s\"" % (str(e), query % (user, host)))
```

Comment:
Exceptions from cursor.execute actually aren't being fully handled properly.  If you look where user_mod is called (around line 508) it only catches SQLParseError and InvalidPrivsError errors.  If cursor.execute throws any other exception (it was throwing a ProgrammingError when I was working on this) then it just crashes ansible. Perhaps the exception handling in the main section needs to be improved upon.